javalib/src/main/scala/java/lang/Math.scala

```diff
@@ -242,6 +242,139 @@ object Math {
       b.toFloat
   }
 
+  def scalb(d: scala.Double, scaleFactor: scala.Int): scala.Double = {
+    // scalastyle:off return
+
+    // Constants
+    val SignBit = scala.Long.MinValue
+    val mbits = 52
+    val ebits = 11
+    val mmask = (1L << mbits) - 1L
+    val emask = (1 << ebits) - 1
+    val minEForSubnormalResult = -mbits - 1 // 1 additional bit for rounding up to MinPositiveValue
```

Comment:
Do I understand correctly that, `val minEForSubnormalResult = -mbits - 1` because `mbits + 1` is the maximum number of bit shifting required for a normalize subnormal numbers here?

```scala
      // Normalize by shifting the leading 1 just out of the mantissa bits, and adjust newE to compensate
      val clz = Long.numberOfLeadingZeros(bits & mmask)
      newE -= (clz - (64 - mbits))
      signAndMantissa = (bits & SignBit) | ((bits << (clz - (64 - mbits - 1))) & mmask)
```

In that case, maybe

```suggestion
    val minEForSubnormalResult = -mbits - 1 // 1 additional bit for rounding up to SmallestPositiveNormalValue
```

?

Reply:
Right, What I was referring to was about the "additional 1-bit shift" during `signAndMantissa` calculation. And here `minEForSubnormalResult` is indeed for rounding to the (subnormal) MinPositiveValue. Got it 👍